lib/kvdb/ikvdb.c

```diff
@@ -1912,15 +1912,22 @@ ikvdb_kvs_names_get(struct ikvdb *handle, size_t *namec, char ***namev)
 {
     struct ikvdb_impl *self = ikvdb_h2r(handle);
     int                i, slot = 0;
-    char **            kvsv;
+    char **            kvsv = NULL;
     char *             name;
 
-    kvsv = calloc(HSE_KVS_COUNT_MAX, sizeof(*kvsv) + HSE_KVS_NAME_LEN_MAX);
-    if (!namev)
-        return merr(ev(ENOMEM));
+    INVARIANT(handle);
+    INVARIANT(namev);
 
     mutex_lock(&self->ikdb_lock);
 
+    /* If no KVSs, short-circuit. */
+    if (!self->ikdb_kvs_cnt)
+        goto out;
+
+    kvsv = calloc(HSE_KVS_COUNT_MAX, sizeof(*kvsv) + HSE_KVS_NAME_LEN_MAX);
+    if (!kvsv)
+        return merr(ev(ENOMEM));
```

Comment:
Release lock before returning error. I would keep allocations outside the critical section if possible.

Reply:
Good catch. I will look at this again